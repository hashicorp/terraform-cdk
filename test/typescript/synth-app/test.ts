// Copyright (c) HashiCorp, Inc
// SPDX-License-Identifier: MPL-2.0
import { TestDriver } from "../../test-helper";

describe("full integration test synth", () => {
  let driver: TestDriver;

  beforeAll(async () => {
    process.env.TF_VAR_myvar =
      "If you see this value we pass runtime information into the synth";
    driver = new TestDriver(__dirname);
    await driver.setupTypescriptProject();
    await driver.synth();
  });

  test("debug command", async () => {
    const { stdout } = await driver.exec(`cdktf debug --json`);
    const { cdktf, constructs } = JSON.parse(stdout);
    expect(cdktf.length).not.toBe(0);
    expect(constructs.length).not.toBe(0);
  });

  test("synth generates JSON", () => {
    expect(driver.synthesizedStack("hello-terra").toString()).toMatchSnapshot();
  });

  test("should not pass TF_VAR environment variables", () => {
    expect(
      driver.synthesizedStack("hello-terra").output("tf-env-var-output")
    ).toBe("no-value-found");
  });
});
