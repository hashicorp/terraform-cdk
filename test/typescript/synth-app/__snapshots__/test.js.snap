// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`full integration test synth synth generates JSON 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"hello-terra\\",
      \\"backend\\": \\"local\\",
      \\"overrides\\": {
        \\"stack\\": [
          \\"terraform\\"
        ],
        \\"aws_sns_topic\\": [
          \\"display_name\\",
          \\"provider\\",
          \\"lifecycle\\"
        ]
      }
    }
  },
  \\"terraform\\": {
    \\"required_providers\\": {
      \\"aws\\": {
        \\"version\\": \\"~> 2.0\\",
        \\"source\\": \\"aws\\"
      }
    },
    \\"backend\\": {
      \\"remote\\": {
        \\"organization\\": \\"test\\",
        \\"workspaces\\": {
          \\"name\\": \\"test\\"
        }
      }
    }
  },
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"region\\": \\"eu-central-1\\",
        \\"ignore_tags\\": {
          \\"keys\\": [
            \\"foo\\"
          ]
        }
      }
    ]
  },
  \\"resource\\": {
    \\"aws_sns_topic\\": {
      \\"Topic\\": {
        \\"display_name\\": \\"topic\\",
        \\"provider\\": \\"aws\\",
        \\"lifecycle\\": {
          \\"create_before_destroy\\": true
        },
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"hello-terra/Topic\\",
            \\"uniqueId\\": \\"Topic\\"
          }
        }
      }
    },
    \\"aws_instance\\": {
      \\"Instance\\": {
        \\"ami\\": \\"ami-12345678\\",
        \\"instance_type\\": \\"t2.micro\\",
        \\"credit_specification\\": {
          \\"cpu_credits\\": \\"standard\\"
        },
        \\"ebs_block_device\\": [
          {
            \\"device_name\\": \\"/dev/sda1\\",
            \\"volume_size\\": 100
          }
        ],
        \\"metadata_options\\": {
          \\"http_endpoint\\": \\"true\\"
        },
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"hello-terra/Instance\\",
            \\"uniqueId\\": \\"Instance\\"
          }
        }
      },
      \\"Instance2\\": {
        \\"ami\\": \\"ami-12345678\\",
        \\"instance_type\\": \\"t2.micro\\",
        \\"credit_specification\\": {
          \\"cpu_credits\\": \\"\${aws_instance.Instance.credit_specification[0].cpu_credits}\\"
        },
        \\"metadata_options\\": {
          \\"http_endpoint\\": \\"\${aws_instance.Instance.metadata_options[0].http_endpoint}\\"
        },
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"hello-terra/Instance2\\",
            \\"uniqueId\\": \\"Instance2\\"
          }
        }
      }
    }
  },
  \\"output\\": {
    \\"snstopicarn\\": {
      \\"value\\": \\"\${aws_sns_topic.Topic.arn}\\"
    },
    \\"instancepassword\\": {
      \\"value\\": \\"\${base64decode(aws_instance.Instance.password_data)}\\"
    },
    \\"instancehttpendpoint\\": {
      \\"value\\": \\"\${aws_instance.Instance.metadata_options[0].http_endpoint}\\"
    }
  }
}"
`;
